### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by ghost**
initial setup of browser and accessibility tests for reference implementation of nlmaps as Embedkaart Standalone.

See readme at this rev for status.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    steps:
+      #install chromium dependencies
+      - run: sudo apt-get update && sudo apt-get install -yq gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget
+
+      - checkout
+
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+          - v1-dependencies-{{ checksum "package.json" }}
```
Comment (boris-arkenaar) [inline .circleci/config.yml]: Wouldn't it be more correct to use `package-lock.json`?

File: scripts/live-server-conf.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (boris-arkenaar) [inline scripts/live-server-conf.json]: Is this being used somewhere now?

File: scripts/serve.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const http = require('http')
+const connect = require('connect')
+const serveStatic = require('serve-static')
+const test = require('../test/test.js');
+
+
+// Simple server for serving static files with test in callback
+const app = connect().use(serveStatic('test/'))
+const server = http.createServer(app).listen(8123, () => {
```
Comment (boris-arkenaar) [inline scripts/serve.js]: Make use of the `live-server-conf.json` for the port?

File: test/test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+const puppeteer = require('puppeteer');
+const tap = require('tap');
+const pa11y = require('pa11y');
+
+async function test(server){
```
Comment (boris-arkenaar) [inline test/test.js]: Nice asinc/await tests like this.
Need to be separated (integration/Aria) when expanded.

File: test/test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    t.assert(div === true, 'div is present');
+    const mapCenter = await page.evaluate(() => map.getCenter());
+    t.assert(mapCenter.lng === 5.9699 && mapCenter.lat=== 52.2112, 'map is centered as expected')
+    await browser.close();
+    t.end()
+  })
+  //ARIA tests
+  await tap.test('there are no accessibility issues', async function(t) {
+    const ariares = await pa11y('http://localhost:8123/index.html', {
+      allowedStandards: 'WCAG2AA',
```
Comment (boris-arkenaar) [inline test/test.js]: Aria config in a separate file somewhere?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,39 @@
...
+-----
+
+### Running directly
+The tests can be run directly with:
+
+    npm run test
+
+During development, you can use e.g. [nodemon](https://github.com/remy/nodemon) to rerun the tests on changes to your test files:
+
+    nodemon --watch scripts --watch test -e js,html scripts/serve.js
```
Comment (boris-arkenaar) [inline README.md]: Make a `npm run test-watch` task for that?

**Review by boris-arkenaar — COMMENTED** [review]
Looks good. It's missing linting at the moment though.

File: test/test.js (hunk 4 above)
Comment (ghost) [reply test/test.js]: Do you mean separate entrypoints/commands for intergration and aria testing?

e.g.:

    npm run test-int
    npm run test-acc

Comment (boris-arkenaar) [reply test/test.js]: Could be, as well. But I meant separation in different files/folders.

File: test/test.js (hunk 5 above)
Comment (ghost) [reply test/test.js]: Good idea. I see from https://github.com/Amsterdam/atlas/blob/develop/test/pa11y.js that my pa11y approach is a  bit naive still. I suppose we can copy the general setup from there?

Comment (boris-arkenaar) [reply test/test.js]: Yes. Do use that!

**Comment by ghost** [comment]
I've addressed most of your points @boris-arkenaar.

* add watch-test command
* add linting (note that the actual nlmaps code is linted separately)
* add docker-compose services for linting and watch-test
* caches in circle-ci use package-lock.json
* remove unused live-server-conf.json. Port can be made configurable later, see next point
* concerning splitting of aria and functional tests, and extracting of aria config, I propose we pick that up in a next iteration when we actually add those tests.

File: test/test.js [hunk 7]
```diff
@@ -2,35 +2,36 @@ const puppeteer = require('puppeteer');
...
+async function test(server) {
+  // General browser tests
+  await tap.test('The map is loaded', async (t) => {
+    const browser = await puppeteer.launch({ args: ['--no-sandbox', '--disable-setuid-sandbox'] });
     const page = await browser.newPage();
     await page.goto('http://localhost:8123/index.html');
-    const div = await page.$eval('#mapdiv', el => el ? true : false);
+    const div = await page.$eval('#mapdiv', el => !!el);
     t.assert(div === true, 'div is present');
+    /* map call occurs in browser context */
```
Comment (boris-arkenaar) [inline test/test.js]: Comments with two slashes (`//`).

File: test/test.js [hunk 8]
```diff
@@ -2,35 +2,36 @@ const puppeteer = require('puppeteer');
...
-  await tap.test('The map is loaded', async function(t) {
-    const browser = await puppeteer.launch({args: ['--no-sandbox', '--disable-setuid-sandbox']});
+async function test(server) {
+  // General browser tests
+  await tap.test('The map is loaded', async (t) => {
+    const browser = await puppeteer.launch({ args: ['--no-sandbox', '--disable-setuid-sandbox'] });
     const page = await browser.newPage();
     await page.goto('http://localhost:8123/index.html');
-    const div = await page.$eval('#mapdiv', el => el ? true : false);
+    const div = await page.$eval('#mapdiv', el => !!el);
```
Comment (boris-arkenaar) [inline test/test.js]: We prefer performing casts explicitly; `Boolean(el)`.

**Review by boris-arkenaar — COMMENTED** [review]
Looks very good!

**Comment by ghost** [comment]
Great, thanks!

**Pull request merged**